luigi/hdfs.py

```diff
@@ -105,6 +105,51 @@ def list_path(path):
         return [path, ]
     return [str(path), ]
 
+def is_dangerous_rm_path(path):
+    """ Finds out if the path is too dangerous to remove.
+    * empty path
+    * / root path
+    * one level deep root path, i.e. /etc or /opt
+    :return boolean, True if too dangerous! Relative path is always ok
+
+    >>> is_dangerous_rm_path(None)
+    True
+    >>> is_dangerous_rm_path('')
+    True
+    >>> is_dangerous_rm_path(' ')
+    True
+    >>> is_dangerous_rm_path('/opt')
+    True
+    >>> try:
+    ...     is_dangerous_rm_path('//')
+    ... except ValueError:
+    ...     pass
+    >>> is_dangerous_rm_path('/opt/')
+    True
+    >>> is_dangerous_rm_path('/opt/foo/bar')
+    False
+    >>> is_dangerous_rm_path('my_relative')
+    False
+    """
+    if not path:
+        return True
```

Comment:
Yea, your worst case for 2) is *completly* unlikely. You mean default would be to delete `/`? And your worst case for 1) is not that bad. Besides. case 1) will not happen. What will happen is that you get an exception thrown from `path.strip()`.

I don't like this line as it's logic that gives no value, and I get confused reading it since I expect lines of code to be useful. :)